### DISCUSSION TITLE
Installation Error - resolve dependencies

### DISCUSSION THREAD

**Discussion by technomudit** [eclipse-syson/syson#77] [Q&A] [answered]
Hi,
I have been trying to install the SysON latest release, and Keep getting the **_attached error log_** when I run the mvn command

**mvn -U -B -e clean verify --settings settings.xml**

```
Could not transfer artifact org.eclipse.sirius:sirius-components-emf:pom:2024.1.4 from/to github-sirius-web (https://maven.pkg.github.com/eclipse-sirius/sirius-web): status code: 401, reason phrase: Unauthorized (401) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project syson-services: Could not resolve dependencies for project org.eclipse.syson:syson-services:jar:2024.1.0: Failed to collect dependencies at org.eclipse.sirius:sirius-components-emf:jar:2024.1.4
```
[debug.txt](https://github.com/eclipse-syson/syson/files/14357156/debug.txt)

**Comment by AxelRICHARD** [comment]
Hello,
I suggest you to run the following command from the root of the repository: `mvn clean verify`.
Before that, you must have configured your `$HOME/.m2/settings.xml` with `github-sirius-emfjson` & `github-sirius-web`.
Your `$HOME/.m2/settings.xml` should look like this:
```
<?xml version="1.0" encoding="UTF-8"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <servers>
    <server>
      <id>github-sirius-emfjson</id>
      <username>$YOUR_GITHUB_USERNAME</username>
      <password>$YOUR_GITHUB_ACCESS_TOKEN</password>
    </server>
    <server>
      <id>github-sirius-web</id>
      <username>$YOUR_GITHUB_USERNAME</username>
      <password>$YOUR_GITHUB_ACCESS_TOKEN</password>
    </server>
  </servers>
</settings>
```
Be sure to replace `$YOUR_GITHUB_USERNAME` with your GitHub user id, and `$YOUR_GITHUB_ACCESS_TOKEN` with the value of your access token.
If you do not have one already, please create a personal token with a scope of `read:package` by following [the GitHub documentation](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token).

Hope it will help you.
Regards,

**Reply by technomudit** [reply]
Dear Axel,
Thanks for your quick response. 
my settings.xml was at C:\Program Files\Maven\apache-maven-3.9.5\conf\settings.xml

I did modify the same as per your suggestion, however I still get the issue.

![image](https://github.com/eclipse-syson/syson/assets/3246618/befb06ea-34e9-4e52-b79f-2c00297a573f)

**Reply by AxelRICHARD** [reply, marked as answer]
Mudit,
By default, your local `settings.xml` file should be in `$HOME/.m2/settings.xml`. If you need to point to a specific `settings.xml` that is not located in the default location, then you should use the following command, in your case:
`mvn clean verify --settings "C:/Program Files/Maven/apache-maven-3.9.5/conf/settings.xml"` should work.
Hope it will help you.
Regards,

**Reply by technomudit** [reply]
It worked !! thanks a lot :)
![image](https://github.com/eclipse-syson/syson/assets/3246618/44b93587-b346-4b4a-853d-ddafc2935933)
